woocommerce/woocommerce-android
# Issue/5933 product images refactor take2

This PR is a partial solution for #5933. My intent was to avoid passing a list of image models in the arguments to `ProductImagesFragment`, but that turned out to be a much larger task than I expected.

There were also other things that needed to be fixed in that fragment - for example, the `viewState` held the image list along with `productImagesState` which contained a copy of the image list, and it also held the list of uploading Uris. So this PR addresses this by moving that data out of the `viewState` and into `LiveData`.

Only one review is required, but I've tagged two developers since you both have a lot of experience with this topic.

To test:

* Go to product detail
* Tap on a product image to open `ProductImagesFragment`
* Ensure it still works as expected, including re-ordering images, uploading images, and viewing a variation's image
